packages/@cdklabs/cdk-cicd-wrapper/src/plugins/security/LambdaDLQPlugin.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as cdk from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as nag from 'cdk-nag';
+import { Construct } from 'constructs';
+import { IPlugin, ResourceContext } from '../../';
+
+export interface LambdaDLQPluginProps {
+  /**
+   * The retentionPeriod for the DLQ.
+   */
+  retentionPeriod?: cdk.Duration;
+}
+
+export class LambdaDLQPlugin implements IPlugin {
+  readonly name = 'LambdaDLQPlugin';
+  readonly version = '1.0';
+
+  readonly retentionPeriod: cdk.Duration;
+
+  constructor(props?: LambdaDLQPluginProps) {
+    this.retentionPeriod = props?.retentionPeriod ?? cdk.Duration.days(14);
+  }
+
+  afterStage(scope: Construct, context: ResourceContext): void {
+    cdk.Aspects.of(scope).add(
+      new LambdaDLQPluginAspect(() => {
+        const dlqStack = new cdk.Stack(scope, `${context.blueprintProps.applicationName}-${this.name}-DLQStack`);
+        const dlq = new sqs.Queue(dlqStack, 'DLQ', {
```

Comment:
maybe you also prefix the 'DLQ' to avoid collisions in same account deployments?